app/pages/cif/project-revision/[projectRevision]/index.tsx

```diff
@@ -58,11 +59,23 @@ export function ProjectRevision({
   const [discardProjectRevision, discardingProjectRevision] =
     useDeleteProjectRevisionMutation();
 
+  const updatedAt = () => {
+    if (query.projectRevision) {
+      return query.projectRevision.updatedAt;
+    }
+    return null;
+  };
+
+  const updatedAtStatic = updatedAt();
+
   const lastEditedDate = useMemo(
-    () => new Date(query.projectRevision.updatedAt),
-    [query.projectRevision.updatedAt]
+    () => (updatedAtStatic ? new Date(updatedAtStatic) : null),
+    [updatedAtStatic]
   );
```

Comment:
Not sure about this part--I needed to check if the project revision existed before trying to `useMemo`, but I can't use `isRedirecting` to do that because it can't come before `useMemo` per hook rules.

Reply:
Also, my unit tests don't cover this part so I only noticed I needed it when I manually checked. Should I do any other tests (integration?) to cover it? This work is unrelated to the new hook, but it is related to whether or not you successfully get the 404 page.